File: lib/models/issue.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'issue.freezed.dart';
part 'issue.g.dart';

enum IssueStatus { pending, approved, rejected }

@freezed
class Issue with _$Issue {
  const factory Issue({
    required int id,
    required String title,
    required String description,
    required String userId,
    required IssueStatus status,
    required DateTime createdAt,
    String? imageUrl,
    List<String>? attachments,
    String? adminNotes,
    DateTime? updatedAt,
    DateTime? resolvedAt,
  }) = _Issue;

  factory Issue.fromJson(Map<String, dynamic> json) => _$IssueFromJson(json);
}

File: app/models/issue.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, ARRAY
from sqlalchemy.orm import relationship
from .user import Base

class Issue(Base):
    __tablename__ = 'issues'
    
    id = Column(Integer, primary_key=True)
    title = Column(String(255), nullable=False)
    description = Column(Text, nullable=False)
    user_id = Column(Integer, ForeignKey('users.id'), nullable=False)
    image_url = Column(String(500))
    attachments = Column(ARRAY(String))
    status = Column(String(50), nullable=False, default='pending')
    admin_notes = Column(Text)
    created_at = Column(DateTime, nullable=False, default=datetime.utcnow)
    updated_at = Column(DateTime, onupdate=datetime.utcnow)
    resolved_at = Column(DateTime)
    
    # Relationship
    user = relationship("User", backref="issues")
    
    def to_dict(self):
        return {
            'id': self.id,
            'title': self.title,
            'description': self.description,
            'image_url': self.image_url,
            'attachments': self.attachments if self.attachments else [],
            'status': self.status,
            'admin_notes': self.admin_notes,
            'created_at': self.created_at.isoformat() if self.created_at else None,
            'updated_at': self.updated_at.isoformat() if self.updated_at else None,
            'resolved_at': self.resolved_at.isoformat() if self.resolved_at else None,
            'username': self.user.username if self.user else None
        } 